source/frontend/src/features/properties/map/acquisition/hooks/useProjectTypeahead.ts

```diff
@@ -0,0 +1,42 @@
+import { useApiProjects } from 'hooks/pims-api/useApiProjects';
+import { IAutocompletePrediction } from 'interfaces';
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+
+interface IProjectTypeahead {
+  handleTypeaheadSearch: (query: string) => Promise<void>;
+  isTypeaheadLoading: boolean;
+  matchedProjects: IAutocompletePrediction[];
+}
+
+export const useProjectTypeahead = (): IProjectTypeahead => {
+  // project type-ahead state
+  const { getProjectPredictions } = useApiProjects();
+  const [isTypeaheadLoading, setIsTypeaheadLoading] = useState(false);
+  const [matchedProjects, setMatchedProjects] = useState<IAutocompletePrediction[]>([]);
+
+  // fetch autocomplete suggestions from server
+  const handleTypeaheadSearch = async (query: string) => {
+    try {
+      setIsTypeaheadLoading(true);
+      const { data } = await getProjectPredictions(query);
+      console.log(data);
+      setMatchedProjects(
```

Comment:
admittedly an edge case, but if the user keeps typing I don't see that we are enforcing search order or handling cancellations here, so in theory if the requests complete out of order you may get results from a previously typed string.

Reply:
Thats right. However this is almost the same code as we have in the other autocomplete and it seems to work fine

Reply:
Yep, you would probably need to force the requests to return in a different order. If you want to leave this as is maybe log some low-priority tech debt? unlikely to see problems from this but strictly speaking the code isn't handling network latency.